packages/flutter/test/widgets/navigator_replacement_test.dart

```diff
@@ -39,42 +42,205 @@ void main() {
+      expect(log, contains('/b pushed, previous route: /a'));
+      expect(log, contains('/a removed, previous route: null'));
+      expect(log, contains('/ removed, previous route: null'));
+
+      observer.onPushed = null;
+      log.clear();
+
+
+      navigator.pushNamed('/');
+      await tester.pumpAndSettle();
+
+      expect(find.text('home', skipOffstage: false), findsOneWidget);
+      expect(find.text('a', skipOffstage: false), findsNothing);
+      expect(find.text('b', skipOffstage: false), findsOneWidget);
+
+      observer
+        ..onPushed = (Route<dynamic> route, Route<dynamic> previousRoute) {
+          expectSync(route is PageRoute && route.settings.name == '/a', isTrue);
+          expectSync(previousRoute is PageRoute && previousRoute.settings.name == '/', isTrue);
+          log.add('/a pushed, previous route: /');
+        }
+        ..onRemoved = (Route<dynamic> route, Route<dynamic> previousRoute) {
+          expectSync(route is PageRoute && route.settings.name == '/', isTrue);
+          expectSync(previousRoute is PageRoute && previousRoute.settings.name == '/b', isTrue);
+          log.add('/ removed, previous route: /b');
+        };
+
+      // Remove only some routes below
+      navigator.pushNamedAndRemoveUntil('/a', ModalRoute.withName('/b'));
+      await tester.pumpAndSettle();
+
+      expect(find.text('home', skipOffstage: false), findsNothing);
+      expect(find.text('a', skipOffstage: false), findsOneWidget);
+      expect(find.text('b', skipOffstage: false), findsOneWidget);
+      expect(log, contains('/a pushed, previous route: /'));
+      expect(log, contains('/ removed, previous route: /b'));
+
+      observer.onPushed = null;
+      log.clear();
+    });
+
+    testWidgets('triggers page transition animation for pushed route', (WidgetTester tester) async {
+
+      final Widget myApp = MaterialApp(
+        home: const Material(child: Text('home')),
+        routes: <String, WidgetBuilder>{
+          '/a': (BuildContext context) => const Material(child: Text('a')),
+          '/b': (BuildContext context) => const Material(child: Text('b')),
+        },
+      );
+
+      await tester.pumpWidget(myApp);
+      final NavigatorState navigator = tester.state(find.byType(Navigator));
+
+      navigator.pushNamed('/a');
+      await tester.pumpAndSettle();
+
+      navigator.pushNamedAndRemoveUntil('/b', (Route<dynamic> route) => false);
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 100));
+
+      // We are mid-transition, both pages are onstage
+      expect(find.text('a'), findsOneWidget);
+      expect(find.text('b'), findsOneWidget);
+
+      // Complete transition
+      await tester.pumpAndSettle();
+      expect(find.text('a'), findsNothing);
+      expect(find.text('b'), findsOneWidget);
+    });
+
+    testWidgets('Hero transition triggers when appropriate', (WidgetTester tester) async {
```

Comment:
instead of "when appropriate" maybe state the condition that this is testing where the hero should fly.